packages/cozy-harvest-lib/src/components/OAuthForm.jsx

```diff
@@ -17,21 +19,39 @@ export class OAuthForm extends PureComponent {
     this.handleAccountId = this.handleAccountId.bind(this)
     this.handleConnect = this.handleConnect.bind(this)
     this.handleOAuthCancel = this.handleOAuthCancel.bind(this)
+    this.handleMoreParams = this.handleMoreParams.bind(this)
     this.state = {
       initialValues: null,
       showingOAuthModal: false
     }
   }
 
   componentDidMount() {
-    const { account } = this.props
+    const { account, konnector, flow, client } = this.props
     this.setState({ initialValues: account ? account.oauth : null })
+
+    const konnectorPolicy = findKonnectorPolicy(konnector)
+
+    if (konnectorPolicy.fetchMoreOAuthUrlParams) {
+      this.setState({ needMoreParams: true })
+      konnectorPolicy.fetchMoreOAuthUrlParams({
+        flow,
+        konnector,
+        client,
+        onSuccess: this.handleMoreParams
```

Comment:
Is there a particular reason here to not use Promises ?

const extraParams = await konnectorPolicy.fetchExtraOAuthUrlParams({...})
this.setState({ extraParams })

Reply:
Right, no good reason